examples/coordinates/plot_sgr-coordinate-frame.py

```diff
@@ -118,71 +118,37 @@ class Sagittarius(coord.BaseCoordinateFrame):
 D = rotation_matrix(SGR_PHI, "z", unit=u.radian)
 C = rotation_matrix(SGR_THETA, "x", unit=u.radian)
 B = rotation_matrix(SGR_PSI, "z", unit=u.radian)
-SGR_MATRIX = np.array(B.dot(C).dot(D))
+A = np.diag([1.,1.,-1.])
+SGR_MATRIX = matrix_product(A, B, C, D)
 
 ##############################################################################
-# This is done at the module level, since it will be used by both the
-# transformation from Sgr to Galactic as well as the inverse from Galactic to
-# Sgr. Now we can define our first transformation function:
+# This rotation matrix is defines at the module level, since it will be used
+# by both the transformation from Sgr to Galactic as well as the inverse from
+# Galactic to Sgr. Now we can define our first transformation function:
 
-@frame_transform_graph.transform(coord.FunctionTransform, coord.Galactic, Sagittarius)
-def galactic_to_sgr(gal_coord, sgr_frame):
-    """ Compute the transformation from Galactic spherical to
+@frame_transform_graph.transform(coord.StaticMatrixTransform, coord.Galactic, Sagittarius)
+def galactic_to_sgr():
+    """ Compute the transformation matrix from Galactic spherical to
         heliocentric Sgr coordinates.
     """
-
-    l = np.atleast_1d(gal_coord.l.radian)
-    b = np.atleast_1d(gal_coord.b.radian)
-
-    X = np.cos(b)*np.cos(l)
-    Y = np.cos(b)*np.sin(l)
-    Z = np.sin(b)
-
-    # Calculate X,Y,Z,distance in the Sgr system
-    Xs, Ys, Zs = SGR_MATRIX.dot(np.array([X, Y, Z]))
-    Zs = -Zs
-
-    # Calculate the angular coordinates lambda,beta
-    Lambda = np.arctan2(Ys,Xs)*u.radian
-    Lambda[Lambda < 0] = Lambda[Lambda < 0] + 2.*np.pi*u.radian
-    Beta = np.arcsin(Zs/np.sqrt(Xs*Xs+Ys*Ys+Zs*Zs))*u.radian
-
-    return Sagittarius(Lambda=Lambda, Beta=Beta,
-                       distance=gal_coord.distance)
+    return SGR_MATRIX
 
 ##############################################################################
-# The decorator ``@frame_transform_graph.transform(coord.FunctionTransform,
+# The decorator ``@frame_transform_graph.transform(coord.StaticMatrixTransform,
 # coord.Galactic, Sagittarius)``  registers this function on the
 # ``frame_transform_graph`` as a coordinate transformation. Inside the function,
-# we simply follow the same procedure as detailed by David Law's `transformation
-# code <http://www.astro.virginia.edu/~srm4n/Sgr/code.html>`_. Note that in this
-# case, both coordinate systems are heliocentric, so we can simply copy any
-# distance from the `~astropy.coordinates.Galactic` object.
+# we simply construct a rotation matrix using the Euler angles (and the
```

Comment:
This is no longer true, maybe move the comment to where the actual matrix is constructed?